src/stdlib_codata_type.fypp

```diff
@@ -0,0 +1,61 @@
+#:include "common.fypp"
+#:set KINDS = REAL_KINDS
+module stdlib_codata_type
+    !! Codata constant type
+    use stdlib_kinds, only: #{for k in KINDS[:-1]}#${k}$, #{endfor}#${KINDS[-1]}$
+    use stdlib_io, only: FMT_REAL_DP
+    use stdlib_optval, only: optval 
+    private
+
+    type, public :: codata_constant
+    !! Derived type for representing a Codata constant.
+        character(len=64) :: name
+        real(dp) :: value
+        real(dp) :: uncertainty
+        character(len=32) :: unit
+    contains 
+        procedure :: print
+        #:for k in KINDS
+        procedure :: to_real_${k}$
+        #:endfor
+        generic :: to_real => #{for k in KINDS[:-1]}#to_real_${k}$, #{endfor}#to_real_${KINDS[-1]}$
+    end type
+
+    interface to_real
+        !! Get the constant value or uncertainty.
+        #:for k in KINDS
+        module procedure to_real_${k}$
+        #:endfor
+    end interface
+
+    public :: to_real
```

Comment:
```suggestion
    public :: real
```
If I am correct, it could be called `real` and follow the same interface as the intrinsic function [`real`](https://gcc.gnu.org/onlinedocs/gfortran/REAL.html)

Reply:
It could be called `real` but I was wondering if it is a good idea to mask an intrinsic function. Moreover, the name suggested by @perazz made sense to me when I looked at the `stdlib_strings` because the name carries the notion of conversion.

Reply:
The intrinsic `real` should not be masked, but just overloaded.
It is also fine with me to use `to_real` (similarly to `to_string`).
@perazz @jalvesz What is your opinion here? do you prefer `real` or `to_real`?

Reply:
Both are fine to me - I also had originally thought about `real`, although the overloaded interface would then be different than the intrinsic: 
- `real(x, KIND)` has `integer, intent(in) :: kind` and the fortran compiler resolves that to the return value
- `to_real(x, MOLD)` has `real(*), intent(in) :: mold` because the return kind cannot otherwise be selected.